mogc_partnerships/pipeline.py

```diff
@@ -8,17 +8,26 @@
 from .models import CohortOffering, Partner, PartnerOffering
 
 
+def user_can_access_course(user, course_key):
+    partner = Partner.objects.get(org=course_key.org)
+    offering = partner.offerings.get(course_key=course_key)
+    cohort_ids = CohortOffering.objects.filter(offering=offering).values_list(
+        "cohort_id", flat=True
+    )
+    user_in_cohort = user.memberships.filter(cohort__in=cohort_ids).exists()
+    user_is_partner_manager = user.management_memberships.filter(
+        user_id=user.id, partner_id=partner.id
+    ).exists()
+    if user_in_cohort or user_is_partner_manager:
+        return True
```

Comment:
@michaelwheeler Curious what you think about where this should live, if it's okay here for now or if it would be better to go ahead and establish a lib.

Reply:
Good question. I think I prefer to leave it here for now. If we find in the future that we are using it all over the place then we can find it a different home.